src/main/java/vazkii/patchouli/client/book/template/component/ComponentImage.java

```diff
@@ -48,14 +59,55 @@ public void render(MatrixStack ms, BookPage page, int mouseX, int mouseY, float
 			return;
 		}
 
+		checkValues();
+
 		page.mc.getTextureManager().bindTexture(resource);
 		ms.push();
 		ms.translate(x, y, 0);
 		ms.scale(scale, scale, scale);
 		RenderSystem.color4f(1F, 1F, 1F, 1F);
 		RenderSystem.enableBlend();
-		DrawableHelper.drawTexture(ms, 0, 0, u, v, width, height, textureWidth, textureHeight);
+		DrawableHelper.drawTexture(ms, 0, 0, uInt, vInt, widthInt, heightInt, textureWidthInt, textureHeightInt);
 		ms.pop();
 	}
 
+	//prepares all the Ivariables and defaults them if they are of improper type
+	private void checkValues() {
+		try {
+			textureWidthInt = textureWidth.asNumber().intValue();
```

Comment:
add a method that returns an optional number and use `orElse` on that instead of using exceptions